File: mobile/lib/features/detection/water_stress_controller.dart
import '../../core/image/picked_image.dart';
import '../../models/water_stress.dart';
import '../../services/water_stress_api.dart';
import 'detection_controller.dart';

/// Crop-water-stress detection controller bound to
/// `POST /api/water-stress/detect`.
class WaterStressController extends DetectionController<WaterStressResult> {
  WaterStressController(WaterStressApi api, ImagePickerFn pickImage)
      : super(
          runner: ({
            required List<int> imageBytes,
            required String imageName,
            String? cropName,
          }) =>
              api.detect(
                imageBytes: imageBytes,
                imageName: imageName,
                cropName: cropName,
              ),
          pickImage: pickImage,
        );
}

File: config/core/controllers/water_stress_controller.py
"""
KisanAI OS
Crop Water Stress Detection Controller

Thin controller for the AI crop-water-stress detection feature.
"""

from config.core.services.water_stress_service import WaterStressService


class WaterStressController:
    """Crop Water Stress Detection Controller"""

    def __init__(self, service=None):
        self.service = service or WaterStressService()

    def detect(self, file, crop_name=None):
        return self.service.detect(file, crop_name)